File: Jardin De infantes/DAL/Servicios/Backup/dalBackup.vb
Imports System.Configuration
Imports System.Data.SqlClient

Public Class dalBackup
    Private db As Conexion
    Private _cmd As IDbCommand
    Sub New()

        db = New Conexion
    End Sub

    Public Function Leer(cadena As String) As DataTable
        Try
            Dim cmd As SqlCommand = db.CrearComando

            cmd.CommandType = CommandType.Text
            cmd.CommandText = cadena
            cmd.ExecuteNonQuery()
            Dim da As New SqlDataAdapter(cmd)
            Dim dt As New DataTable
            da.Fill(dt)
            db.RealizarCommit()

            Return dt

        Catch ex As Exception
            db.RealizarRollBack()
            MsgBox(ex.Message, MsgBoxStyle.Critical, "Error")
            Return Nothing
        End Try
        db.cerrarConexion()
    End Function

    Public Sub Escribir(cadena As String)
        Try

            Dim cmd As SqlCommand = db.CrearComando

            cmd.CommandType = CommandType.Text
            cmd.CommandText = cadena

            cmd.ExecuteNonQuery()

            db.RealizarCommit()

        Catch ex As Exception
            db.RealizarRollBack()
            MsgBox(ex.Message, MsgBoxStyle.Critical, "Error")

        End Try

    End Sub
    Public Sub EscribirsinTran(cadena As String)
        Try

            Using cn As New SqlConnection(ConfigurationManager.ConnectionStrings("Notebook").ConnectionString)
                cn.Open()

                Dim cmd As New SqlCommand()
                cmd.Connection = cn
                cmd.CommandType = CommandType.Text
                cmd.CommandText = cadena

                cmd.ExecuteNonQuery()


            End Using


        Catch ex As Exception

            MsgBox(ex.Message, MsgBoxStyle.Critical, "Error")

        End Try

    End Sub
End Class

File: Jardin De infantes/Seguridad/DALseguridad/Backup/dalBackup.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Data.SqlClient;
using System.Configuration;
using MPPSeguridad;
using System.Data;
using System.Windows.Forms;

namespace DALseguridad.Backup
{

    public class dalBackup
    {
       
        DataTable dt = new DataTable();
        SqlTransaction tx;
        public DataTable Leer(string texto)
        {
            try
            {
                using (SqlConnection cn = new SqlConnection(dalConexion.Cadena))
                {
                    cn.Open();
                    tx = cn.BeginTransaction();
                    SqlCommand cmd = new SqlCommand(texto,cn,tx);
                    cmd.CommandType = CommandType.Text;
                    cmd.ExecuteNonQuery();
                    SqlDataAdapter da = new SqlDataAdapter(cmd);
                    da.Fill(dt);
                    tx.Commit();
                    cn.Close();
                    return dt;
                }
            }
            catch (Exception e)
            {
                tx.Rollback();
                MessageBox.Show("Error en el Backup",e.Message ,
                                MessageBoxButtons.OK, MessageBoxIcon.Error); ;
                return null;
            }
           
        }
 
        public void Escribir(string texto)
        {
            try
            {
                using (SqlConnection cn = new SqlConnection(dalConexion.Cadena))
                {
                    cn.Open();
                    tx = cn.BeginTransaction();
                    SqlCommand cmd = new SqlCommand(texto, cn, tx);
                    cmd.CommandType = CommandType.Text;
                    cmd.ExecuteNonQuery();
                    tx.Commit();

                }
            }
            catch (Exception e)
            {
                tx.Rollback();
                MessageBox.Show("Error en el Backup", e.Message,
                                MessageBoxButtons.OK, MessageBoxIcon.Error); ;
               
            }
        }
        public void EscribirSinTX(string texto)
        {
            try
            {
                using (SqlConnection cn = new SqlConnection(dalConexion.Cadena))
                {
                    cn.Open();
                    
                    SqlCommand cmd = new SqlCommand(texto, cn, tx);
                    cmd.CommandType = CommandType.Text;
                    cmd.ExecuteNonQuery();
                    

                }
            }
            catch (Exception e)
            {
               
                MessageBox.Show("Error en el Backup", e.Message,
                                MessageBoxButtons.OK, MessageBoxIcon.Error); ;

            }
        }
    }
}
